### PR TITLE
search and tooltip improvements in Create Unit Dialog

### PR DESCRIPTION
**Pull request by HUSSAR-mtrela**
- added searching units by their categories (e.g. ANTIAIR)
- added searching units by their weapon labels (e.g. AutoCannon)
- added tooltip to all filters
- added config options to hide Visibility and Source filters
- added new filters Dummy, Playable/Buildable units, Testing (Paragon + SCU + ACU + T3 Engineers)
- moved blueprint logic to a new file in prep to share it with rest of UI elements
- moved Visibility filters to a new row
- fixed issue with saving filter presets
- fixed layout and resizing issues
- fixed issue #4822 (2 checkboxes next to Visibility in old UI)

image

### CODE REVIEW

File: lua/ui/dialogs/createunit.lua [hunk 1]
```diff
@@ -536,34 +565,39 @@ GetNameFilters = {
                     {
                         title = 'Construction',
                         key = 'const',
+                        tooltip = '<LOC spawn_filter_const>Toogle construction units',
```
Comment (Garanas) [inline lua/ui/dialogs/createunit.lua:568]: 'Toogle' instead of 'Toggle'

Comment (HUSSAR-mtrela) [reply lua/ui/dialogs/createunit.lua]: it is fixed now.

**Comment by Garanas** [comment]
@HUSSAR-mtrela I'll review it this evening, further changes can be deferred to a new pull request.

**Comment by Garanas** [comment]
There's a typo:

image

The save / delete / clear all buttons feel a bit miss aligned (at 125% UI scaling):

image

Beyond that it looks good! Could you elaborate why you moved the anonymous functions to a separate file? No need to change it back, just trying to understand

**Comment by Garanas** [comment]
@HUSSAR-mtrela I think the unit creation dialog is where it should be now 😄

**Comment by HUSSAR-mtrela** [comment]
@Garanas thanks for making improvements and merging this PR.
I moved the anonymous functions to a separate file because at some point most of these functions could be moved to utility module that could be shared between other UI elements and even mods.

**Pull request merged**